spec/lib/profiler_spec.rb

```diff
@@ -198,4 +198,27 @@ def self.bar(baz, boo)
       expect(Rack::MiniProfiler.snapshots_transporter?).to eq(true)
     end
   end
+
+  describe '#call' do
+    let(:app) { lambda { |env| [200, {}, ["OK"]] } }
+    let(:profiler) { Rack::MiniProfiler.new(app) }
+
+    it "returns well-formed response when stackprof isn't installed" do
+      response = profiler.call({ "PATH_INFO" => "/", "QUERY_STRING" => "pp=flamegraph" })
+      status, headers, body = response
+
+      expect(100..599).to include(status)
+      expect(headers.keys).to(all(be_a(String)))
+      expect(body).to(all(be_a(String)))
```

Comment:
Rather than check for an Enumerable of Strings, lets do the Rack-spec thing and check for response to each?

Reply:
Done. This caught a bug in the memory profiler spec, so thanks for that!